content/chainguard/administration/cloudevents/image-copy-gcr/index.md

```diff
@@ -0,0 +1,196 @@
+    * If you don't know your organization's UIDP, you can retrieve it by running `chainctl iam organizations list -o table`.
+* The next line specifies the location of the Artifact Registry repository and the Cloud Run subscriber.
+* The final line defines `dst_repo` value, which is used to create a name for the repository in the Artifact Registry where images will be mirrored.
+
+As an example, if the `name` value you specify is `chainguard-dev` and the `dst_repo` value is `mirrored` (as shown in the following example) any pushes to `cgr.dev/<organization>/foo` will be mirrored to `<location>-docker.pkg.dev/<project_id>/chainguard-dev-mirrored`
+
+Be sure to include to include a closing curly bracket after the final line.
+
+```
+  name = "chainguard-dev"
+
+  project_id = "<project-id>"
+
+  group = "<organization-id>"
+
+  location = "us-central1" 
+
+  dst_repo = "mirrored"
+}
+```
+
+You can create this file with a command like the following.
+
+```sh
+cat > main.tf <<EOF
+module "image-copy" {
+  source = "github.com/chainguard-dev/platform-examples//image-copy-gcp/iac"
+
+  name = "chainguard-dev"
+
+  project_id = "<project-id>"
+
+  group = "<organization-id>"
+
+  location = "us-central1" 
+
+  dst_repo = "mirrored"
+}
+EOF
+```
+
+After that you can apply the configuration to get the application up and running.
+
+
+## Applying the configuration
+
+The Terraform configuration you created in the previous section will do all the work of setting up an events subscription for you. 
+
+Specifically, it will build the mirroring application into an image using `ko_build` and deploy the app to a Cloud Run service with permission to push to the Google Artifact Registry. It also sets up a Chainguard Identity with permissions to pull from the private `cgr.dev` repository, allows the Cloud Run service's service account  to assume the puller identity, and sets up a subscription to notify the Cloud Run service when pushes happen to `cgr.dev`.
+
+Before applying the configuration, you'll need to log in to both the Chainguard platform with `chainctl` and GCP with `gcloud`.
+
+```sh
+chainctl auth login
+gcloud auth login
+```
+
+If you haven't already done so, you will also need to acquire new access credentials to use as [Application Default Credentials](https://cloud.google.com/docs/authentication/application-default-credentials). You can do so with this command.
+
+```sh
+gcloud auth application-default login
+```
+
+Following that, run `terraform init` to initialize Terraform’s working directory.
+
+```sh
+terraform init
+```
+
+Then run `terraform plan`. This will produce a speculative execution plan that outlines what steps Terraform will take to create the resources defined in the file you set up in the last section.
+
+```sh
+terraform plan
+```
+
+If the plan worked successfully and you’re satisfied that it will produce the resources you expect, you can apply it.
+
+```sh
+terraform apply
+```
```

Comment:
I got a warning from this command:

```
│ Warning: Redundant empty provider block
│ 
│   on .terraform/modules/image-copy/image-copy-gcp/iac/main.tf line 18:
│   18: provider "chainguard" {}
│ 
│ Earlier versions of Terraform used empty provider blocks ("proxy provider configurations") for child modules to declare their
│ need to be passed a provider configuration by their callers. That approach was ambiguous and is now deprecated.
│ 
│ If you control this module, you can migrate to the new declaration syntax by removing all of the empty provider "chainguard"
│ blocks and then adding or updating an entry like the following to the required_providers block of module.image-copy:
│     chainguard = {
│       source = "chainguard-dev/chainguard"
│     }
│ 
│ (and 2 more similar warnings elsewhere)
```

Is it something we could update to not have this deprecation warning? otherwise it might be worth adding a note about it

My terraform version:
```
Terraform v1.8.4
on linux_amd64
+ provider registry.terraform.io/chainguard-dev/chainguard v0.1.8
+ provider registry.terraform.io/chainguard-dev/cosign v0.0.20
+ provider registry.terraform.io/hashicorp/google v5.31.1
+ provider registry.terraform.io/ko-build/ko v0.0.15
```

Reply:
Pretty sure I was seeing this warning as well. Good call, I think I see where the upstream fix is for this. I'll resolve this convo but know that I'll play around with fixing this asap, along with the other upstream fix